src/main.js

```diff
@@ -50,9 +52,48 @@ const exportedUtils = Object.assign(publicUtils, {
  * > [Apify platform](../guides/apify-platform) and require your scripts to be authenticated.
  * > See the [authentication guide](../guides/apify-platform#logging-into-apify-platform-from-apify-sdk) for instructions.
  *
+ * ## `Apify` Class
+ *
+ * As opposed to those helper functions, there is an alternative approach using `Apify` class (a named export).
+ * It has mostly the same API, but the methods on `Apify` instance will use the configuration provided in the constructor.
+ * Environment variables are will have precedence over this configuration.
+ *
+ * ```js
+ * const { Apify } = require('apify-js'); // use named export to get the class
+ *
+ * const sdk = new Apify({ token: '123' });
+ * console.log(sdk.config.get('token')); // '123'
+ *
+ * // the token will be passed to the `call` method automatically
+ * const run = await sdk.call('apify/hello-world', { myInput: 123 });
+ * console.log(`Received message: ${run.output.body.message}`);
+ * ```
+ *
+ * Another example shows how the default dataset name can be changed:
+ *
+ * ```js
+ * const { Apify } = require('apify-js'); // use named export to get the class
+ *
+ * const sdk = new Apify({ defaultDatasetId: 'custom-name' });
+ * await sdk.pushData({ myValue: 123 });
```

Comment:
Hmm, something we did not consider is how this will work on the platform. The `pushData` includes `openDataset` so in theory, it should create the dataset with this name, if it does not exist and then use it.

Reply:
Not sure I follow here either. `sdk.pushData()` will call `sdk.openDataset()`, which will open the default dataset - and this sdk instance have the default dataset name configured to `custom-name`. 

As opposed to the code underneath, where the dataset name needs to be passed in the `openDataset()` call - but it is the same name as in the sdk instance example.

Both examples are calling `openDataset('custom-name')` under the hood (the latter one uses the global helper function, but it is the very same code).

Reply:
Yeah. The thing is, on the platform, the UI and API are tied to the `defaultDatasetId` that's provided by the platform in the env var. When you set the `defaultDatasetId` in the actor, it will not change the real platform `defaultDatasetId`. So the UI will display no results and if you call `actorRun.get` API call you will get a `defaultDatasetId` that does not point to the one you configured.

I start to think if it's a good idea at all to allow users to configure the default storage IDs. On one hand, they could've done it with `process.env.APIFY_DEFAULT_DATASET_ID = 'stuff'` already. But that's obscure and undocumented. If we document it and make examples for it, I'm pretty sure a lot of people will break their actors with this.